### PR TITLE
Add codespell config, action + fixed typos

### PR DESCRIPTION
**Pull request by yarikoptic**
Saw opentelemetry-go use codespell, checked out opentelemetry - saw that this repo doesn't use it.

### CODE REVIEW

File: content/en/blog/2022/frontend-overhaul/index.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ covering each of the following OTel-supported languages:
 - [Rust](/docs/instrumentation/rust/)
 
 Every microservice has a specific goal and can communicate with others by using
-a global gRPC definition. Persistent information is saved into a PostgresSQL
+a global gRPC definition. Persistent information is saved into a postgresql
```
Comment (jaronoff97) [inline content/en/blog/2022/frontend-overhaul/index.md:67]: i think postgres should remain capitalized

**Review by jaronoff97 — APPROVED** [review]

File: content/en/blog/2022/frontend-overhaul/index.md (hunk 1 above)
Comment (yarikoptic) [reply content/en/blog/2022/frontend-overhaul/index.md:67]:
```suggestion
a global gRPC definition. Persistent information is saved into a postgreSQL
```

**Comment by cartermp** [comment]
@svrnm @chalin what are your thoughts and feels here?

File: content/en/docs/instrumentation/php/manual.md [hunk 2]
```diff
@@ -395,7 +395,7 @@ called on the meter provider.
 ### Asynchronous meters
 
 Async meters are `observable`, eg `ObservableGauge`. When registering an
-observable/async meter, you provide one ore more callback functions. The
+observable/async meter, you provide one or more callback functions. The
```
Comment (yarikoptic) [inline content/en/docs/instrumentation/php/manual.md:398]: I guess that is what  cased somehow

```
[warn] content/en/docs/instrumentation/php/manual.md
[warn] Code style issues found in the above file. Forgot to run Prettier?
[help] Run: npm run format
Error: Process completed with exit code 1.
```
????

Comment (svrnm) [reply content/en/docs/instrumentation/php/manual.md:398]: yeah, it's probably one char to width, run `npm run format` and do a new commit to fix this

**Comment by svrnm** [comment]
Thanks for your PR @yarikoptic !

> @svrnm @chalin what are your thoughts and feels here?

Fun fact: I was trying out [misspell](https://github.com/client9/misspell) the other day, since the [java repo is using it](https://github.com/open-telemetry/opentelemetry-java/blob/main/.github/workflows/reusable-misspell-check.yml), I was not aware that go has yet another tool ;-)

On this PR: First of all I would think that we should separate the content changes from the tooling changes. @yarikoptic: can you put everything which touches files in content/en/docs/... into a new PR and remove it from this one, so we can take in those changes while having the discussion on the tooling.

On the tooling:
We have some setup for cSpell in place already, there is also some stuff for textlint I added additionally to make sure certain things are written in the right way (open source without a hyphen, OpenTelemetry, K8s, gRPC, etc.), so we should be careful by adding yet another tool in that domain. Saying that, I personally have no issue to use something better, so if codespell can do all of that + give us easy PR checks for spelling (i.e. [by copying that from the go repo](https://github.com/open-telemetry/opentelemetry-go/blob/main/.github/workflows/codespell.yaml)) I am in.

Finally, I want to hear @chalin's point of view, since he has the best overview and insights on the infra.

**Comment by yarikoptic** [comment]
> @yarikoptic: would you be open to give the cspell-action as an alternative a try?

Unfortunately not - but feel welcome to just close this PR in favor of some other. Butt if you decide to go with codespell, I can shape up PR to your liking. Just let me know final verdict/instructions

**Comment by svrnm** [comment]
As discussed during the SIG call we have agreed that we don't want to add a spell checking github action right now, because it imposes more requirements to contributors to satisfy and we want to keep the bar low for people to contribute. What would be great is some tooling that we can use post-merge or on a regular basis to scan the repo for spelling issues and fix them.

Apologies for the back-and-forth @yarikoptic

**Review by svrnm — APPROVED** [review]

**Comment by yarikoptic** [comment]
> What would be great is some tooling that we can use post-merge or on a regular basis to scan the repo for spelling issues and fix them.

FWIW: sounds like a suggested codespell action just ran on cron on `main` instead of PRs ;-)

**Pull request merged**